packages/protocol/contracts/authorizers/ClientWhitelister.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "./RankedAdminnable.sol";
+import "./interfaces/IClientWhitelister.sol";
+
+/// @title Authorizer contract where clients are whitelisted until an
+/// expiration time or indefinitely (until the whitelisting is revoked)
+abstract contract ClientWhitelister is RankedAdminnable, IClientWhitelister {
```

Comment:
Why is this contract abstract? I only ask because my understanding is that contracts are identified as abstract contracts when at least one of their functions lacks an implementation. Maybe the idea is that this contract never gets deployed and be always used as a base contract for others to inherit.

Reply:
Actually it is `ClientRrpAuthorizer` that needs to be abstract because it doesn't implement `AUTHORIZER_TYPE`. I just copy pasted it to create `ClientWhitelisted` so that ended up being an abstract as well. Fixed that now.